views/default/input/button.php

```diff
@@ -9,15 +9,10 @@
  * @uses $vars['class'] Additional CSS class
  */
 
-if (isset($vars['class'])) {
-	$vars['class'] = "elgg-button {$vars['class']}";
-} else {
-	$vars['class'] = "elgg-button";
-}
+$vars['class'] = (array) elgg_extract('class', $vars, []);
+$vars['class'][] = "elgg-button";
 
-$defaults = array(
-	'type' => 'button',
-);
+$defaults = ['type' => 'button'];
```

Comment:
Would anyone pass a custom type?
Perhaps just $vars['type'] and force it to button